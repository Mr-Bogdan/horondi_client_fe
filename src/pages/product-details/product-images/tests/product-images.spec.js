import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ProductImages from '../product-images';

jest.mock('connected-react-router', () => {
  jest.fn();
});
jest.mock('../product-images.styles', () => ({ useStyles: () => ({}) }));
jest.mock('react-redux');
const mockUseContext = jest.fn().mockImplementation(() => ({
  isLight: true
}));
React.useContext = mockUseContext;

const dispatch = jest.fn();
const storage = {
  language: 0,
  images: {
    primary: {
      large: '',
      medium: '',
      small: ''
    },
    additional: []
  }
};

useDispatch.mockImplementation(() => dispatch);
useSelector.mockImplementation(() => storage);

describe('ProductImages page test', () => {
  it('Should render ProductImages', () => {
    const component = shallow(<ProductImages images={{
      primary: { large: '' },
      additional: [{ large: '' }]
    }} />);
    expect(component).toBeDefined();
  });
});
