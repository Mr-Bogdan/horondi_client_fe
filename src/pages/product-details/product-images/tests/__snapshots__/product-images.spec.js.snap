// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImages page test Should match snapshot 1`] = `
<div>
  <ImgsViewer
    closeBtnTitle="Закрити"
    currImg={0}
    enableKeyboardInput={true}
    imgCountSeparator=" / "
    imgs={Array []}
    isOpen={false}
    leftArrowTitle="Попередній"
    onClickNext={[Function]}
    onClickPrev={[Function]}
    onClickShowNextImg={true}
    onClickThumbnail={[Function]}
    onClose={[Function]}
    preloadNextImg={true}
    preventScroll={true}
    rightArrowTitle="Наступний"
    showCloseBtn={true}
    showImgCount={true}
    showThumbnails={true}
    spinner={[Function]}
    spinnerColor="#fff"
    spinnerSize={50}
    theme={Object {}}
    thumbnailOffset={2}
    width={1024}
  />
  <div>
    <div />
    <div>
      <ZoomImage
        images={
          Object {
            "additional": Array [],
            "primary": Object {
              "large": "",
              "medium": "",
              "small": "",
            },
          }
        }
      />
    </div>
  </div>
</div>
`;
